components/clp-package-utils/clp_package_utils/scripts/native/search.py

```diff
@@ -116,6 +117,30 @@ async def worker_connection_handler(reader: asyncio.StreamReader, writer: asynci
     return worker_connection_handler
 
 
+def _get_ipv4_address() -> str | None:
+    """
+    Retrieves an IPv4 address of the host for network communication.
+
+    :returns: The first non-loopback IPv4 address it finds.
+    If no non-loopback address is available, returns the first loopback IPv4 address.
+    If no IPv4 address is found, returns None.
+    """
+    fallback_ip = None
+
+    for _, addresses in psutil.net_if_addrs().items():
+        for addr in addresses:
+            if addr.family == socket.AF_INET:
+                ip = addr.address
+                if not ipaddress.ip_address(ip).is_loopback:
+                    return ip
+                if fallback_ip is None:
+                    fallback_ip = ip
```

Comment:
Early continue makes the code more readable.
```suggestion
            if addr.family != socket.AF_INET:
                continue
            ip = addr.address
            if not ipaddress.ip_address(ip).is_loopback:
                return ip
            if fallback_ip is None:
                fallback_ip = ip
```